refactored/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+//! Findex is a cryptographic algorithm allowing to securely maintain an
+//! encrypted index.
+//!
+//! It uses a generic Dictionary Encryption Scheme (Dx-Enc) as building block to
+//! implement a Graph Encryption Scheme (Gx-Enc) which allows indexing both
+//! `Location`s and `Keyword`s.
+//!
+//! The `Index` traits is not a cryptographic algorithm but is used to simplify
+//! the interface and to hide the cryptographic details of the implementation
+//! when it is possible.
+//!
+//! However, since Findex is database agnostic and is generic on the choice of
+//! the Dx-Enc algorithm, some generic parameters are still exposed through the
+//! `Index` interface.
+
+#![feature(never_type)]
+#![allow(incomplete_features)]
+#![feature(generic_const_exprs)]
```

Comment:
as discussed, is this feature finally required?

Reply:
removed in 9712ff0